src/ovp_pipeline/context_binder.py

```
@@ -0,0 +1,679 @@
+       See :func:`build_retrieval_context`.
+
+    ``system_prompt_body`` is what the handler concatenates after
+    BL-075's USER + RULES prefix (loaded inline here when the
+    handler hasn't already attached it; safe to dedup downstream).
+
+    Raises :class:`ValueError` for unknown ``anchor_kind``.
+    """
+    if anchor_kind not in ANCHOR_KINDS:
+        raise ValueError(
+            f"unknown anchor kind {anchor_kind!r}; " f"expected one of {sorted(ANCHOR_KINDS)}"
+        )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)